src/vs/code/electron-main/windows.ts

```diff
@@ -376,6 +380,28 @@ export class WindowsManager implements IWindowsMainService {
 		}
 
 		let configuration: IWindowConfiguration;
+		let openInNewWindow = openConfig.preferNewWindow || openConfig.forceNewWindow;
+
+		// Restore any existing backup workspaces
+		if (openConfig.restoreBackups) {
+			const workspacesWithBackups = this.backupService.getWorkspaceBackupPaths();
+
+			workspacesWithBackups.forEach(workspacePath => {
+				if (!fs.existsSync(workspacePath)) {
+					this.backupService.removeWorkspaceBackupPathSync(Uri.file(workspacePath));
+					return;
+				}
+
+				const untitledToRestore = this.backupService.getWorkspaceUntitledFileBackupsSync(Uri.file(workspacePath)).map(filePath => {
```

Comment:
@Tyriar same comment here about how to restore untitled files. This would not be needed if untitled had input factory.